From 167e195bf1af429c106ba782a546ff10b870679a Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Fri, 27 Sep 2013 16:48:31 +0300
Subject: [PATCH 374/987] dpa_offload: Propagate error code when IPSec module
 fails to recognize ioctl command

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

IPSec module used to silently fail when an ioctl command was not
recognized. This may cause applications to take inconsistent (uninitialized)
data as granted.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Change-Id: Id2ec3b0894d9d80332c279c976127c1d40345c7e
Reviewed-on: http://git.am.freescale.net:8181/5038
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Varvara Andrei-B21317 <andrei.varvara@freescale.com>
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
---
 drivers/staging/fsl_dpa_offload/wrp_dpa_ipsec.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/wrp_dpa_ipsec.c b/drivers/staging/fsl_dpa_offload/wrp_dpa_ipsec.c
index 0d3ff7e..7b51e97 100644
--- a/drivers/staging/fsl_dpa_offload/wrp_dpa_ipsec.c
+++ b/drivers/staging/fsl_dpa_offload/wrp_dpa_ipsec.c
@@ -1335,7 +1335,8 @@ free:
 		break;
 	}
 	default:
-		pr_err("Invalid DPA IPsec ioctl\n");
+		pr_err("Invalid DPA IPsec ioctl (0x%x)\n", cmd);
+		ret = -EINVAL;
 		break;
 	}
 
@@ -1503,7 +1504,8 @@ free:
 		break;
 	}
 	default:
-		pr_err("Invalid DPA IPsec ioctl\n");
+		pr_err("Invalid DPA IPsec ioctl (0x%x)\n", cmd);
+		ret = -EINVAL;
 		break;
 	}
 
-- 
1.9.1

